src/logic/contracts/safeContracts.ts

```diff
@@ -20,24 +21,39 @@ export const DEFAULT_FALLBACK_HANDLER_ADDRESS = '0xd5D82B6aDDc9027B22dCA772Aa68D
 export const SAFE_MASTER_COPY_ADDRESS_V10 = '0xb6029EA3B2c51D09a50B53CA8012FeEB05bDa35A'
 
 
-let proxyFactoryMaster
-let safeMaster
-
-const createGnosisSafeContract = (web3: Web3) => {
-  const gnosisSafe = contract(GnosisSafeSol)
-  gnosisSafe.setProvider(web3.currentProvider)
-
-  return gnosisSafe
+let proxyFactoryMaster: GnosisSafeProxyFactory
+let safeMaster: GnosisSafe
+
+/**
+ * Creates a Contract instance of the GnosisSafe contract
+ * @param {Web3} web3
+ * @param {ETHEREUM_NETWORK} networkId
+ */
+const createGnosisSafeContract = (web3: Web3, networkId: ETHEREUM_NETWORK) => {
```

Comment:
Should this be typed to return `GnosisSafe` ?

Reply:
the returned type is inferred in this case, shall we add it anyway? I truly don't remember.

Reply:
Yes you are right, I think it's not needed